src/internal/operations/backup.go

```diff
@@ -154,11 +163,139 @@ func (op *BackupOperation) Run(ctx context.Context) (err error) {
 	return err
 }
 
+// calls kopia to retrieve prior backup manifests, metadata collections to supply backup heuristics.
+func produceManifestsAndMetadata(
+	ctx context.Context,
+	kw *kopia.Wrapper,
+	sw *store.Wrapper,
+	sel selectors.Selector,
+	acct account.Account,
+) ([]*snapshot.Manifest, []data.Collection, error) {
+	complete, closer := observe.MessageWithCompletion("Fetching backup heuristics:")
+	defer func() {
+		complete <- struct{}{}
+		close(complete)
+		closer()
+	}()
+
+	m365, err := acct.M365Config()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var (
+		tid         = m365.AzureTenantID
+		oc          = selectorToOwnersCats(sel)
+		ms          = kopia.FetchPrevSnapshotManifests(ctx, kw.Conn(), &oc)
+		collections []data.Collection
+	)
+
+	for _, man := range ms {
+		bup := backup.Backup{}
+		if err := sw.Get(
+			ctx,
+			model.BackupSchema,
+			model.StableID(man.Tags[kopia.BackupIDTag]),
+			&bup,
+		); err != nil {
+			return nil, nil, err
+		}
+
+		for _, fn := range []string{graph.DeltaTokenFileName} {
```

Comment:
nit: eventually we should have a function in the producer that we call and it gives us a set of metadata file names (or partial paths, or maybe even complete paths). That will isolate generating the paths from consuming them, making the logic here more generic

ex.
```go
  for _, fn := range graph.GetMetadataPaths() {
```